src/index.js

```diff
@@ -133,29 +134,32 @@ class Knob extends Component<KnobProps, KnobState> {
 
   // converts a value to an angle for the knob respecting the additional rotateDegrees prop
   convertValueToAngle(value: number): number {
-    const angle = (this.scale(value) + this.props.rotateDegrees) % 360;
+    const angle = (this.getScale()(value) + this.props.rotateDegrees) % 360;
     return this.normalizeAngle(angle);
   }
 
-
   // converts an angle to a value for the knob respecting the additional rotateDegrees prop
   convertAngleToValue(angle: number): number {
     const angle2 = angle - this.props.rotateDegrees;
-    return this.scale.invert(this.normalizeAngle(angle2))
+    return this.getScale().invert(this.normalizeAngle(angle2))
   }
 
-  setupScaling(min, max, clampMin, clampMax) {
-    // scaling of min/max to degree values needs to be swapped when turning the knob by 180 degree, to have the minimum values on the left side
-    if (this.props.rotateDegrees < 270 && this.props.rotateDegrees > 90) {
-      this.scale = scaleLinear()
-        .domain([min, max])
-        .range([clampMin, clampMax]);
-    } else {
-      this.scale = scaleLinear()
-        .domain([max, min])
-        .range([clampMax, clampMin]);
+  getScale() {
```

Comment:
The pattern here is that we have some internal state (`this.scale`) that we only want to recalculate when the props change. The react team calls this "memoization": https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization